// @vx/shape/esm/shapes/LineRadial.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';
import { radialLine } from 'd3-shape';
LineRadial.propTypes = {
  className: PropTypes.string,
  children: PropTypes.func,
  curve: PropTypes.func,
  data: PropTypes.any,
  defined: PropTypes.oneOfType([PropTypes.func, PropTypes.bool]),
  fill: PropTypes.string,
  angle: PropTypes.oneOfType([PropTypes.func, PropTypes.number]),
  radius: PropTypes.oneOfType([PropTypes.func, PropTypes.number]),
  innerRef: PropTypes.oneOfType([PropTypes.func, PropTypes.object])
};
export default function LineRadial(_ref) {
  var className = _ref.className,
      angle = _ref.angle,
      radius = _ref.radius,
      defined = _ref.defined,
      curve = _ref.curve,
      data = _ref.data,
      innerRef = _ref.innerRef,
      children = _ref.children,
      _ref$fill = _ref.fill,
      fill = _ref$fill === void 0 ? 'transparent' : _ref$fill,
      restProps = _objectWithoutPropertiesLoose(_ref, ["className", "angle", "radius", "defined", "curve", "data", "innerRef", "children", "fill"]);

  var path = radialLine();
  if (angle) path.angle(angle);
  if (radius) path.radius(radius);
  if (defined) path.defined(defined);
  if (curve) path.curve(curve);
  if (children) return children({
    path: path
  });
  return React.createElement("path", _extends({
    ref: innerRef,
    className: cx('vx-line-radial', className),
    d: path(data),
    fill: fill
  }, restProps));
}
