// @vx/hierarchy/esm/hierarchies/Pack.js
import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { Group } from '@vx/group';
import { pack as d3pack } from 'd3-hierarchy';
import DefaultNode from '../HierarchyDefaultNode';
Pack.propTypes = {
  root: PropTypes.object.isRequired,
  children: PropTypes.func,
  top: PropTypes.number,
  left: PropTypes.number,
  className: PropTypes.string,
  radius: PropTypes.func,
  size: PropTypes.arrayOf(PropTypes.number),
  padding: PropTypes.number,
  nodeComponent: PropTypes.any
};
export default function Pack(_ref) {
  var top = _ref.top,
      left = _ref.left,
      className = _ref.className,
      root = _ref.root,
      radius = _ref.radius,
      size = _ref.size,
      padding = _ref.padding,
      children = _ref.children,
      _ref$nodeComponent = _ref.nodeComponent,
      nodeComponent = _ref$nodeComponent === void 0 ? DefaultNode : _ref$nodeComponent;
  var pack = d3pack();
  if (size) pack.size(size);
  if (radius !== undefined) pack.radius(radius);
  if (padding) pack.padding(padding);
  var data = pack(root);
  if (children) return children(data);
  return React.createElement(Group, {
    top: top,
    left: left,
    className: cx('vx-pack', className)
  }, nodeComponent && data.descendants().map(function (node, i) {
    return React.createElement(Group, {
      key: "pack-node-" + i
    }, React.createElement(nodeComponent, {
      node: node
    }));
  }));
}
