// @vx/hierarchy/esm/hierarchies/Partition.js
import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { Group } from '@vx/group';
import { partition as d3partition } from 'd3-hierarchy';
import DefaultNode from '../HierarchyDefaultNode';
Partition.propTypes = {
  root: PropTypes.object.isRequired,
  children: PropTypes.func,
  top: PropTypes.number,
  left: PropTypes.number,
  className: PropTypes.string,
  size: PropTypes.arrayOf(PropTypes.number),
  round: PropTypes.bool,
  padding: PropTypes.number,
  nodeComponent: PropTypes.any
};
export default function Partition(_ref) {
  var top = _ref.top,
      left = _ref.left,
      className = _ref.className,
      root = _ref.root,
      size = _ref.size,
      round = _ref.round,
      padding = _ref.padding,
      children = _ref.children,
      _ref$nodeComponent = _ref.nodeComponent,
      nodeComponent = _ref$nodeComponent === void 0 ? DefaultNode : _ref$nodeComponent;
  var partition = d3partition();
  if (size) partition.size(size);
  if (round) partition.round(round);
  if (padding) partition.padding(padding);
  var data = partition(root);
  if (children) return children(data);
  return React.createElement(Group, {
    top: top,
    left: left,
    className: cx('vx-partition', className)
  }, nodeComponent && data.descendants().map(function (node, i) {
    return React.createElement(Group, {
      key: "partition-node-" + i
    }, React.createElement(nodeComponent, {
      node: node
    }));
  }));
}
